### PR TITLE
commands: explain change in command description splitting

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

A comment here would help the reader to understand the need for this splitting logic, which is not so straightforward.

Addresses review comment in https://github.com/Homebrew/brew/pull/15146#discussion_r1157361656.

Also, apply the same logic to shell commands. This was missed in #15146.

### CODE REVIEW

File: Library/Homebrew/commands.rb [hunk 1]
```diff
@@ -203,6 +203,10 @@ def command_description(command, short: false)
 
     if (cmd_parser = Homebrew::CLI::Parser.from_cmd_path(path))
       if short
+        # We only consider a dot as a full stop if it is either followed by a
+        # whitespace or at the end of the description. In this way we can
+        # prevent cutting off a sentence in the middle due to dots in URLs or
+        # paths.
         cmd_parser.description.split(/\.(?>\s|$)/).first
```
Comment (MikeMcQuaid) [inline Library/Homebrew/commands.rb:210]: Given the regex is reused: can it be split to a constant or variable?

**Review by MikeMcQuaid — COMMENTED** [review]
Great work so far!

File: Library/Homebrew/commands.rb [hunk 2]
```diff
@@ -32,6 +32,11 @@ module Commands
     "lc"          => "livecheck",
     "tc"          => "typecheck",
   }.freeze
+  # This pattern is used to split descriptions at full stops. We only consider a
+  # dot as a full stop if it is either followed by a whitespace or at the end of
+  # the description. In this way we can prevent cutting off a sentence in the
+  # middle due to dots in URLs or paths.
+  DESCRIPTION_SPLITTING_PATTERN = /\.(?>\s|$)/.freeze
```
Comment (ZhongRuoyu) [inline Library/Homebrew/commands.rb:39]: @MikeMcQuaid Thanks for the suggestion! How about this?

**Review by MikeMcQuaid — APPROVED** [review]
Looks great, thanks @ZhongRuoyu!

**Pull request merged**